app/src/main/java/me/ccrama/redditslide/Activities/CommentsScreenSingle.java

```diff
@@ -209,70 +211,26 @@ public void onPageScrollStateChanged(int state) {
     boolean archived;
     boolean contest;
 
-    private class AsyncGetSubredditName extends AsyncTask<String, Void, String> {
+    public void update(Submission s, String subreddit) {
+        this.subreddit = subreddit;
+        locked = s.isLocked();
+        archived = s.isArchived();
+        contest = s.getDataNode().get("contest_mode").asBoolean();
+        setupAdapter();
+    }
 
-        @Override
-        protected void onPostExecute(String s) {
-            subreddit = s;
-            setupAdapter();
+    @Override
+    public void onPause() {
+        super.onPause();
+        if (asyncGetSubredditName != null) {
+            asyncGetSubredditName.cancel(true);
```

Comment:
onPause doesn't seem like the right place for this - it would cancel if a user switches apps temporarily, I imagine it should be in onDestroy

Reply:
If we cancel it `onDestroy()` then there is no surety to invoke the method.  What about doing it in `onStop()` method? If you are sure about this then I can change it to `onDestroy()` .

Reply:
If the app is killed before onDestroy happens (e.g. After a pause) the whole process goes, which includes any asynctasks. So there's no worry of leaking that way

Reply:
Okay, I get it now. I am going to make those changes for all the activities and then commit those changes.